### PR TITLE
feat(browser-client): deleteConversation function

### PR DESCRIPTION
**Pull request by Trinculo54**
Adds the function from the cURL example in **Issue #342** along with an example of the function in the file ```use-browser-client.js``` for people to understand when to use it... *(if they need to)*

### CODE REVIEW

File: src/ChatGPTBrowserClient.js [hunk 1]
```diff
@@ -178,6 +178,71 @@ export default class ChatGPTBrowserClient {
...
+            },
+
+            body: JSON.stringify({
+                is_visible: false,
+            }),
+        };
+        const { debug } = this.options;
+
+        // eslint-disable-next-line no-async-promise-executor
+        const response = await new Promise(async (resolve, reject) => {
```
Comment (waylaidwanderer) [inline src/ChatGPTBrowserClient.js:199]: IMO you can just return `new Promise...` as people can `await deleteConversation` anyway.

File: src/ChatGPTBrowserClient.js [hunk 2]
```diff
@@ -178,6 +178,71 @@ export default class ChatGPTBrowserClient {
...
+            try {
+                let done = false;
+                await fetchEventSource(`${url}/${conversationID}`, {
```
Comment (waylaidwanderer) [inline src/ChatGPTBrowserClient.js:202]: This should just use `fetch` as it's not an event stream response. I copied an example network request from my devtools:

```
await fetch("${url}/${conversationID}", {
    "credentials": "include",
    "headers": {
        "User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:109.0) Gecko/20100101 Firefox/113.0",
        "Accept": "*/*",
        "Accept-Language": "en-US,en;q=0.5",
        "Content-Type": "application/json",
        "Authorization": "Bearer <TOKEN>",
        "Sec-Fetch-Dest": "empty",
        "Sec-Fetch-Mode": "cors",
        "Sec-Fetch-Site": "same-origin",
        "Pragma": "no-cache",
        "Cache-Control": "no-cache"
    },
    "referrer": "https://chat.openai.com/?model=text-davinci-002-render-sha",
    "body": "{\"is_visible\":false}",
    "method": "PATCH"
});
```

File: src/ChatGPTBrowserClient.js [hunk 3]
```diff
@@ -178,6 +178,71 @@ export default class ChatGPTBrowserClient {
         return response;
     }
 
+    async deleteConversation(conversationID) {
```
Comment (waylaidwanderer) [inline src/ChatGPTBrowserClient.js:181]: I'd prefer this to be `conversationId` instead of `conversationID` to match the other parts of the code.

**Review by waylaidwanderer — CHANGES_REQUESTED** [review]

**Comment by Trinculo54** [comment]
Changes should be pushed in the new commit.

**Pull request merged**